src/ResourceManagement/Azure/Azure.Tests/BatchTests.cs

```diff
@@ -60,9 +167,76 @@ public class BatchTests
             {
-                Assert.True(false, "Should not have caught any exception - " + ex);
+                try
+                {
+                    var resourceManager = CreateResourceManager();
+                    resourceManager.ResourceGroups.Delete(rgName);
+                }
+                catch { }
+            }
+        }
+
+        [Fact(Skip = "TODO: Convert to recorded tests")]
+        public async Task CanCreateBatchAccountWithApplication()
+        {
+            try
+            {
+                var applicationId = "myApplication";
+                var applicationDisplayName = "displayName";
+                var allowUpdates = true;
+
+                var batchManager = CreateBatchManager();
+
+                // Create
+                var batchAccount = await batchManager.BatchAccounts
+                    .Define(batchAccountName)
+                    .WithRegion(Region.ASIA_SOUTHEAST)
+                    .WithNewResourceGroup(rgName)
+                    .DefineNewApplication(applicationId)
+                        .WithDisplayName(applicationDisplayName)
+                        .WithAllowUpdates(allowUpdates)
+                        .Attach()
+                    .WithNewStorageAccount(storageAccountName)
+                    .CreateAsync();
+                Assert.Equal(rgName, batchAccount.ResourceGroupName);
+                Assert.NotNull(batchAccount.AutoStorage);
+                Assert.Equal(ResourceUtils.NameFromResourceId(batchAccount.AutoStorage.StorageAccountId), storageAccountName);
+
+                // List
+                var accounts = batchManager.BatchAccounts.ListByGroup(rgName);
+                var found = false;
+                foreach (var account in accounts)
+                {
+                    if (StringComparer.OrdinalIgnoreCase.Equals(account.Name, batchAccountName))
+                    {
+                        found = true;
+                    }
+                }
+
+                Assert.True(found);
+                // Get
+                batchAccount = batchManager.BatchAccounts.GetByGroup(rgName, batchAccountName);
+                Assert.NotNull(batchAccount);
+
+                Assert.True(batchAccount.Applications().ContainsKey(applicationId));
+                var application = batchAccount.Applications()[applicationId];
+
+                Assert.NotNull(application);
+                Assert.Equal(application.DisplayName, applicationDisplayName);
+                Assert.Equal(application.UpdatesAllowed, allowUpdates);
+
+                try
+                {
+                    await batchManager.BatchAccounts.DeleteAsync(batchAccount.ResourceGroupName, batchAccountName);
+                }
+                catch
+                {
+                }
+                var batchAccounts = batchManager.BatchAccounts.ListByGroup(rgName);
+
+                Assert.Equal(batchAccounts.Count, 0);
             }
             finally
             {
```

Comment:
CreateBatchManager() can be moved to https://github.com/Azure/azure-sdk-for-net/blob/Fluent/src/ResourceManagement/Azure/Azure.Tests/TestHelper.cs